### ISSUE TITLE
Typescript port?

### ISSUE THREAD

**Issue by Kilowhisky** [meggsimum/geoserver-node-client#119]
Any interest in a typescript port? Or at least a typescript `module.d.ts` ?

**Comment by Kilowhisky** [comment]
My port is sitting here: https://github.com/Kilowhisky/geoserver-node-client/tree/typescript 
Should basically be good to go, i just need to get bundling working.

**Comment by chrismayer** [comment]
Hi @Kilowhisky, thanks for your interest in the `geoserver-node-client` project. I am not really the Typescript guy, but for me it sounds like offering `module.d.ts` file(s) would be the less invasive way to provide Typescript support, right?

I won't exclude the possibility to port over to TS at all, but that would need some planning due to project management (versions and backwards compatibility, etc.). 

So long story short: Having `module.d.ts` file(s) in the near future would be cool. Would you be willing to provide those?
